pkg/controller/customresource/protection.go

```diff
@@ -0,0 +1,66 @@
+package customresource
+
+import (
+	"context"
+	"encoding/json"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+)
+
+const (
+	AnnotationLastAppliedConfiguration = "mongodb.com/last-applied-configuration"
+)
+
+type OperatorChecker func(resource mdbv1.AtlasCustomResource) (bool, error)
+type AtlasChecker func(resource mdbv1.AtlasCustomResource) (bool, error)
+
+func IsOwner(resource mdbv1.AtlasCustomResource, protectionFlag bool, operatorChecker OperatorChecker, atlasChecker AtlasChecker) (bool, error) {
+	if !protectionFlag {
+		return true, nil
+	}
+
+	wasManaged, err := operatorChecker(resource)
+	if err != nil {
+		return false, err
+	}
```

Comment:
If it was managed we do not need to go to atlas right? we can just return true here.

Reply:
Make sense